awscli/customizations/s3/filegenerator.py

```diff
@@ -116,7 +116,7 @@ class FileGenerator(object):
     ``FileInfo`` objects to send to a ``Comparator`` or ``S3Handler``.
     """
     def __init__(self, client, operation_name, follow_symlinks=True,
-                 page_size=None, result_queue=None):
+                 page_size=None, result_queue=None, request_parameters=None):
```

Comment:
Maybe? If I add it, I will probably will want to break the ``BucketLister`` interface to use ``request_parameters`` in lieu of the current arguments it takes. How about I add ``page_size`` to ``request_parameters`` whenever we decide that we need to use ``request_parameters`` because of a needed extra request parameter for ``ListObjects``?